### ISSUE TITLE
Is fuse.js suitable: compare 100 predetermined questions to an entered question?

### ISSUE THREAD

**Issue by MCF3389** [krisk/Fuse#172]
Hi everyone,

I just wanted to know whether you think that fuse.js was suitable to compare a question entered
by a person with 100 predetermined questions. I've played a bit with fuse's.js online tool and results got
increasingly worse when I reached over 40 predetermined questions. However, I'm right now not at the peak of my programming skills and just wanna know whether this approach will be suitable (or whether I would need something like natural language processing to get good results, which I cannot do unfortunately as a psychologist with programming skills of a beginner ;-)) and maybe what I can do to improve my search. 

One of the questions could be:
`     {
        question: "At what time have you started suffering from pains?",
        keywords: {
          kw_one: "complaints",
          kw_two: "since when",
          kw_three: "begin",
          kw_four: "How long"
        }
     }`

Keys:
`["question","keywords.kw_one", "keywords.kw_two", "keywords.kw_three", "keywords.kw_four"]`

Basically the search engine should list this as first proposed question, even if a person types
"How long have you had complaints?"

Here are my options:
`var options = {
  shouldSort: true,
  tokenize: true,
  matchAllTokens: true,
  threshold: 0.3,
  location: 0,
  distance: 100,
  maxPatternLength: 64,
  minMatchCharLength: 1,
  keys: [
    "question",
    "keywords.kw_one",
    "keywords.kw_two",
    "keywords.kw_three",
    "keywords.kw_four"
]
};
var fuse = new Fuse(list, options); // "list" is the item array
var result = fuse.search("");
`

Should I use weighted search in order to improve my results? And can I add for one question 4 key words, while I add for other classes 6-10 Keywords?

Best regards :)
Max

**Comment by kavika13** [comment]
The question, as formulated, is a difficult unsolved problem. If you need very good results, then either be prepared to do a manual pass after the automated pass, or be prepared to start a bleeding edge tech company.

https://www.ibm.com/watson/services/natural-language-classifier/

If you just need a little basic sorting, and perfection isn't an issue, then a solution that you (or I) program might have better results than nothing at all.
